ArchiSteamFarm/Bot.cs

```diff
@@ -635,6 +649,80 @@ internal sealed class Bot : IDisposable {
 			return await ArchiWebHandler.GetTradeHoldDurationForTrade(tradeID).ConfigureAwait(false);
 		}
 
+		internal async Task<OrderedDictionary> GetUsedKeys() {
+			if (!File.Exists(KeysToRedeemUsedFilePath)) {
+				return new OrderedDictionary();
+			}
+
+			try {
+				OrderedDictionary usedKeys = new OrderedDictionary();
+
+				using (StreamReader reader = new StreamReader(KeysToRedeemUsedFilePath)) {
+					string line;
+
+					while ((line = await reader.ReadLineAsync().ConfigureAwait(false)) != null) {
+						if (line.Length == 0) {
+							continue;
+						}
+
+						string[] parsedArgs = line.Split(DefaultBackgroundKeysRedeemerSeparator, StringSplitOptions.RemoveEmptyEntries);
+						if (parsedArgs.Length < 4) {
+							ArchiLogger.LogGenericWarning(string.Format(Strings.ErrorIsInvalid, line));
+							continue;
+						}
+
+						string status = parsedArgs[1].Substring(1, parsedArgs[1].Length - 2);
+						string key = parsedArgs[3];
+
+						usedKeys[key] = status;
+					}
+				}
+
+				return usedKeys;
+			}
+			catch (Exception e) {
+				ArchiLogger.LogGenericException(e);
+				return null;
+			}
+		}
+
+		internal async Task<OrderedDictionary> GetUnusedKeys() {
```

Comment:
**[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)**

Reply:
really should have done it when i thought "naaaah" 😂